python/ray/util/sgd/data/utils.py

```diff
@@ -0,0 +1,9 @@
+def get_arr_partition(arr, n, i):
+    per_part = len(arr) // n
+    extra = len(arr) % n
```

Comment:
Having un-even shards might influence training behavior I think. Feel free to correct me here, I'm not certain. If I'm right, it might be important to make using the extra data an explicit choice.

Reply:
I don't think shard size should matter because we do a synchronous sgd. This partitioning keeps all partitions within +/- 1 of each other so it should perform well (perhaps batch size alignment of shards could be a little better?). 

Making the shard sizes even would influence training behavior because it would cause non uniform sampling.

Reply:
Alright, looks good then :+1: